examples/twittermap/web/public/javascripts/sidebar/controllers.js

```diff
@@ -1,16 +1,31 @@
 angular.module("cloudberry.sidebar", ["cloudberry.common"])
-  .controller("SidebarCtrl", function($scope, $timeout, cloudberry, moduleManager, cloudberryClient, queryUtil, cloudberryConfig) {
-
+  .controller("SidebarCtrl", function($scope, $timeout, cloudberry, moduleManager, cloudberryClient, queryUtil, cloudberryConfig, $http) {
+    
     // Flag whether current result is outdated
     $scope.isHashTagOutdated = true;
     $scope.isSampleTweetsOutdated = true;
-
     // Flag whether sidebar tab is open
     $scope.isHashTagOpen = false;
-    $scope.isSampleTweetsOpen = true;
-
+    $scope.isSampleTweetsOpen = false;
     $scope.currentTab = "sampletweetTab";
-
+    var sampleTweets = [];
+    $scope.drawTweetMode = 2; //Initially set to 2 liveTweets Mode, change this variable to 1 to enable traditional draw
+    var timeRange = 3; // Set length of time interval in seconds
+    var sendQueryLoop = {}; //Store variable for window.setInterval function, keep only one setInterval function avtive at a time
+    $scope.liveTweetsLoop = {};//Store variable for window.setInterval function, stop live tweets feeding when user specified a 
+                            //time interval in time bar
+
+    var secondLiveTweetQueryTimeOut = null;
+    var timeSeriesEnd = new Date(cloudberry.parameters.timeInterval.end);// This date will be the latest date of tweets been ingested
```

Comment:
Here what you mean is the configured end date of the whole time bar when it's initialized. which should be the logic: `cloudberry.endDate ? cloudberry.endDate : new Date()`(the same as line 341 in `common/services.js`). So, you have an assumption here which is this file must be run after `service.js` run. But it seems a little confused, because line 157 you defined another `timeBarMax` variable using the same assignment statement which will confuse others that these two variables should always equal to each other. So, let's avoid this confusion to later developers by copying initialization logic `cloudberry.endDate ? cloudberry.endDate : new Date()` here.